engines/strict_authentication/app/controllers/strict_authentication/authentication_controller.rb

```diff
@@ -24,11 +24,12 @@ def path_allowed?(path)
       has_sles11 = @system.products.where(identifier: 'SUSE_SLES').first
       return true if (has_sles11 && (path =~ %r{/12/} || path =~ %r{/12-SP1/}))
 
-      @system.repositories.pluck(:local_path).each do |allowed_path|
-        return true if path =~ /^#{Regexp.escape(allowed_path)}/
-      end
+      all_allowed_paths.find { |allowed_path| path =~ /^#{Regexp.escape(allowed_path)}/ }
+    end
 
-      false
+    def all_allowed_paths
```

Comment:
Please add a comment describing the use case why all versions get added to the allowed paths here, this is not obvious.

Reply:
Also a test for this new feature would be good I think.

Reply:
> Also a test for this new feature would be good I think.

I will address the tests in a newer PR so that we can get this merged and on time for the release for the SUMA team